### PR TITLE
Document Auth SDK

### PR DESCRIPTION
**Pull request by TommyJ1994**
## Motivation
Document how to add the Cordova SDK (IDM) to a Cordova App and use it so developers have instructions on how to use the SDK

### Description
Create Documentation for the Auth SDK.

### CODE REVIEW

**Comment by wtrocki** [comment]
@TommyJ1994 How about integrating this change and building on top of that later?

File: docs/auth/README.adoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+=== Initializing the SDK
+`AuthService` can be retrieved by importing it:
+----
+import { AuthService } from 'aerogear/auth';
+----
+Before the `AuthService` can be used, the `init` function must be invoked once in an app.
+----
+let authService = new AuthService(keycloakConfig)
+authService.init()
```
Comment (StephenCoady) [inline docs/auth/README.adoc]: .init takes the onLoad config

Comment (wtrocki) [reply docs/auth/README.adoc]: @TommyJ1994 I have been changed api slightly after integration with core parser. Let's merge that change (see separate PR) and we can update this one.

**Comment by TommyJ1994** [comment]
@StephenCoady @ciaranRoche Updated the PR with some more docs if you have time to look?

File: docs/auth/README.adoc [hunk 2]
```diff
@@ -0,0 +1,123 @@
+= AeroGear Services Auth SDK
+
+Mobile authentication SDK based on link:http://www.keycloak.org/[Keycloak] using link:http://openid.net/connect/[OpenID Connect].
+
+== Usage
+
+To use the Auth SDK you'll first need to:
+
+* Have a Keycloak instance. See the <<Setting up Keycloak>> section.
+* Import the Core module. See link:./core/README.adoc[the Core documentation].
```
Comment (ciaranRoche) [inline docs/auth/README.adoc]: Do we still need the core module imported?

File: docs/auth/README.adoc [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+* If you do not have mobile services enabled in your openshift cluster follow this link:https://github.com/aerogear/mobile-core/blob/master/docs/walkthroughs/local-setup.adoc[Local Setup] guide.
+* Navigate to your Openshift cluster and in the Service Catalog search for the Keycloak service.
+* Click on the Keycloak service and you will be prompted to fill in details about your app.  For now you can leave these as they are.  Navigate through the setup and click Create.
+This will provision the Keycloak service in the project you specified and create a public Client to be used with an app along with a bearer-only client.
+See link:http://www.keycloak.org/docs/latest/server_admin/index.html#oidc-clients[Keycloak OIDC client documentation].
+
+After provisioning, the Keycloak service will be available at the exposed Route. You can view this route inside your project or use the below command to get the route:
+----
+oc get route keycloak --template "http://{{.spec.host}}"
```
Comment (wtrocki) [inline docs/auth/README.adoc]: This is no longer actual. People should use mobile-cli to get mobile-services.json file etc.

File: docs/auth/README.adoc (hunk 2 above)
Comment (wtrocki) [reply docs/auth/README.adoc]: No. Core README will be gone in @paolobueno PR.

**Comment by TommyJ1994** [comment]
@wtrocki @ciaranRoche I removed the unrequired sections - see https://github.com/aerogear/aerogear-js-sdk/pull/39/commits/b137b5f0f552cfabbd368bce9ea86d5ea06e1641

**Comment by wtrocki** [comment]
This breaks on IOS. We could merge that and I will fix that in separate PR or we could hold it. Leaving up to you guys.

**Comment by wtrocki** [comment]
@TommyJ1994 I think this is missing classic mobile-cli documentation where people generate config etc. We are still mentioning keycloak standalone. AeroGear-js-SDK will work now only with the mobile-cli config. There is no standalone support. If that is an requirement we may revise the code.

File: docs/auth/README.adoc [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+* If you do not have mobile services enabled in your openshift cluster follow this link:https://github.com/aerogear/mobile-core/blob/master/docs/walkthroughs/local-setup.adoc[Local Setup] guide.
+* Navigate to your Openshift cluster and in the Service Catalog search for the Keycloak service.
+* Click on the Keycloak service and you will be prompted to fill in details about your app.  For now you can leave these as they are.  Navigate through the setup and click Create.
+This will provision the Keycloak service in the project you specified and create a public Client to be used with an app along with a bearer-only client.
+See link:http://www.keycloak.org/docs/latest/server_admin/index.html#oidc-clients[Keycloak OIDC client documentation].
+
+==== Standalone
+
+To setup standalone Keycloak follow Keycloak's guide link:/https://github.com/keycloak/keycloak/blob/master/README.md[here].
```
Comment (wei-lee) [inline docs/auth/README.adoc]: are the above section the same as in https://github.com/aerogear/aerogear-android-sdk/blob/master/docs/modules/getting-started/pages/auth.adoc?

Comment (TommyJ1994) [reply docs/auth/README.adoc]: Yes, all 3 Auth SDK's have the some repeated text at the top as there is no shared docs space at the moment from what I can see. :(

Comment (wei-lee) [reply docs/auth/README.adoc]: @finp any suggestions? are you working on a shared docs repo yet?

File: docs/auth/README.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    });
+----
+
+You can pass `login-required` or `check-sso` to the init function. `login-required` will authenticate the client if the user is logged-in to Keycloak or display the login page if not.
+`check-sso` will only authenticate the client if the user is already logged-in, if the user is not logged-in the browser will be redirected back to the application and remain unauthenticated.
+
+To enable either of these, just pass the config to the init method:
+
+```
+authService.init({onLoad: "check-sso"})
```
Comment (wtrocki) [inline docs/auth/README.adoc]: No longer needed to pass "check-sso". We are doing that on the library level. If you feel that's wrong we going to remove it from the code.

File: docs/auth/README.adoc [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    }).catch((err) => console.error("Error retrieving user profile", err));
+} else {
+    // not authenticated
+}
+----
+
+=== Logging Out
+To logout, invoke the `logout` function:
+----
+authService.logout()
```
Comment (wtrocki) [inline docs/auth/README.adoc]: I will add handler for promise for better information.

File: docs/auth/README.adoc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+*Note*: Initialising the `AuthService` will also perform the authentication action.
+
+*Note*: In AngularJS, it is necessary to initialise the Auth SDK before bootstrapping Angular to prevent Redirect looping issues after authentication.
+An example of how to perform this in Angular 4 is available in the Aerogear Cordova Showcase App's https://github.com/aerogear/cordova-showcase-template/blob/master/src/app/main.ts[main.ts].
+
+=== Manual Authentication
+To redirect to the login screen, invoke the `login` function:
+----
+authService.login()
```
Comment (wtrocki) [inline docs/auth/README.adoc]: I will add handler for promise for better information.

**Comment by TommyJ1994** [comment]
@wtrocki I updated the init options example to use `login-required` in the example, and mentioned `check-sso` is the default.

File: docs/auth/README.adoc (hunk 4 above)
Comment (finp) [reply docs/auth/README.adoc]: any generic topics should be pushed to mobile-docs, with the following in mind:
* not yet, I need to merge my antora config changes
* not applicable for just sections
Let me get back to you on this, as it's an important subject and needs a structured approach

**Pull request merged**